runtime/dart_init.cc

```diff
@@ -306,7 +309,11 @@ Dart_Isolate IsolateCreateCallback(const char* script_uri,
                                              std::move(jni_class_provider));
 #endif
 
-    if (!snapshot_data.empty()) {
+    if (kernel_data && !kernel_data.empty()) {
+      // We are running kernel code.
+      FTL_CHECK(!LogIfError(Dart_LoadKernel(Dart_ReadKernelBinary(kernel_data.data(),
```

Comment:
`Dart_ReadKernelBinary` - transforms a kernel blob into a program object
`Dart_LoadKernel` - loads that program object for execution in the VM

I'm not sure why the VM team split what was one snapshot call into two separate kernel calls. Perhaps so that additional transforms can be performed on the program object before it is executed.